src/main/java/io/sealights/plugins/sealightsjenkins/utils/JenkinsUtils.java

```diff
@@ -96,4 +98,27 @@ public String getWorkspace(AbstractBuild<?, ?> build) {
         String workingDir = ws.getRemote();
         return workingDir;
     }
+
+    public String resolveCurrentJobName(Descriptor descriptor) throws UnsupportedEncodingException {
+        String descFullUrl = descriptor.getDescriptorFullUrl();
+        URLDecoder.decode(descFullUrl,"UTF-8");
+        String resolvedJobName = descFullUrl.substring(0, descFullUrl.lastIndexOf("/descriptor"));
+        resolvedJobName = resolvedJobName.substring(resolvedJobName.lastIndexOf("/") + 1);
+        return resolvedJobName;
+    }
+
+    public Job getCurrentJob(Descriptor descriptor){
+        try {
+            List<Job> items = Jenkins.getInstance().getItems(Job.class);
+            String currentJobName = resolveCurrentJobName(descriptor);
+            for (Job j : items) {
+                if (j.getName().equals(currentJobName)) {
+                    return j;
+                }
+            }
+        }catch (Exception e){
+
```

Comment:
Don't catch exception and sallow it. Either log it or throw as RuntimeException.